starfish/pipeline/features/spots/detector/detect.py

```diff
@@ -0,0 +1,211 @@
+from starfish.intensity_table import IntensityTable
+from starfish.munge import dataframe_to_multiindex
+from starfish.pipeline.features.spot_attributes import SpotAttributes
+from starfish.typing import Number
+
+
+def measure_spot_intensity(
+        image: Union[np.ndarray, xr.DataArray],
+        spots: pd.DataFrame,
+        measurement_function: Callable[[Sequence], Number]) \
+        -> pd.Series:
+    """measure the intensity of each spot in spots in the corresponding image
+
+    Parameters
+    ----------
+    image : Union[np.ndarray, xr.DataArray],
+        3-d volume in which to measure intensities
+    spots : pd.DataFrame
+        SpotAttributes table containing coordinates and radii of spots
+    measurement_function : Callable[[Sequence], Number]) \
+        Function to apply over the spot volumes to identify the intensity (e.g. max, mean, ...)
+
+    Returns
+    -------
+    pd.Series :
+        Intensities for each spot in SpotAttributes
+
+    """
+
+    def fn(row: pd.Series) -> Number:
+        int_row: pd.Series = row.astype(int)
+        result = measurement_function(
+            image[
+                int_row['z_min']:int_row['z_max'],
+                int_row['y_min']:int_row['y_max'],
+                int_row['x_min']:int_row['x_max']
+            ]
+        )
+        return result
+
+    return spots.apply(
+        fn,
+        axis=1
+    )
+
+
+def measure_spot_intensities(
+        data_image: ImageStack,
+        spot_attributes: pd.DataFrame,
+        measurement_function: Callable[[Sequence], Number], ) \
+        -> IntensityTable:
+    """given spots found from a reference image, find those spots across a data_image
+
+    Parameters
+    ----------
+    data_image : ImageStack
+        ImageStack containing multiple volumes for which spots' intensities must be calculated
+    spot_attributes : pd.Dataframe
+        Locations and radii of spots
+    measurement_function : Callable[[Sequence], Number]) \
+        Function to apply over the spot volumes to identify the intensity (e.g. max, mean, ...)
+
+    Returns
+    -------
+    IntensityTable :
+        3d tensor of (spot, channel, round) information for each coded spot
+
+    """
+
+    n_ch = data_image.shape[Indices.CH]
+    n_round = data_image.shape[Indices.ROUND]
+    spot_attribute_index = dataframe_to_multiindex(spot_attributes)
+    image_shape: Tuple[int, int, int] = data_image.raw_shape[2:]
+    intensity_table = IntensityTable.empty_intensity_table(
+        spot_attribute_index, n_ch, n_round, image_shape)
+
+    indices = product(range(n_ch), range(n_round))
+    for c, r in indices:
+        image, _ = data_image.get_slice({Indices.CH: c, Indices.ROUND: r})
+        blob_intensities: pd.Series = measure_spot_intensity(
+            image, spot_attributes, measurement_function)
+        intensity_table[:, c, r] = blob_intensities
+
+    return intensity_table
+
+
+def concatenate_spot_attributes_to_intensities(
```

Comment:
When calling spots in each (channel, round) sequentially, a method is needed to stitch all the resulting SpotAttributes tables together. This is that method.